### PR TITLE
8297934: [BACKOUT] Compiler should only use verified interface types for optimization

### PR DESCRIPTION
**Pull request by rwestrel**
Backout that change due to hard to reproduce failures and some
failures with stress options. This includes the backout of JDK-8297556
and JDK-8297343 which are fixes to the initial change.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297934](https://bugs.openjdk.org/browse/JDK-8297934): [BACKOUT] Compiler should only use verified interface types for optimization

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11450/head:pull/11450` \
`$ git checkout pull/11450`

Update a local copy of the PR: \
`$ git checkout pull/11450` \
`$ git pull https://git.openjdk.org/jdk pull/11450/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11450`

View PR using the GUI difftool: \
`$ git pr show -t 11450`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11450.diff">https://git.openjdk.org/jdk/pull/11450.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/ProblemList.txt [hunk 1]
```diff
@@ -68,6 +68,9 @@ compiler/c2/Test8004741.java 8235801 generic-all
 
 compiler/codecache/jmx/PoolsIndependenceTest.java 8264632 macosx-all
 
+compiler/debug/TestStressCM.java 8297343 generic-all
+compiler/debug/TestStressIGVNAndCCP.java 8297343 generic-all
+
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/ProblemList.txt:73]: This should be removed.

**Review by TobiHartmann — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/ProblemList.txt (hunk 1 above)
Comment (rwestrel) [reply test/hotspot/jtreg/ProblemList.txt]: Right. Done.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Review by vnkozlov — APPROVED** [review]
Good. Waiting testing results.

**Comment by TobiHartmann** [comment]
Testing is all clean.

**Pull request closed without merging**